server/schedulers/hot_region.go

```diff
@@ -746,65 +770,42 @@ func (bs *balanceSolver) checkDstByPriorityAndTolerance(maxLoad, expect *statist
 // calcProgressiveRank calculates `bs.cur.progressiveRank`.
 // See the comments of `solution.progressiveRank` for more about progressive rank.
 func (bs *balanceSolver) calcProgressiveRank() {
-	src := bs.cur.srcStore
-	dst := bs.cur.dstStore
-	srcLd := src.LoadPred.Min()
-	dstLd := dst.LoadPred.Max()
 	bs.cur.progressiveRank = 0
-	peer := bs.cur.srcPeerStat
 
 	if toResourceType(bs.rwTy, bs.opTy) == writeLeader {
-		if !bs.isTolerance(src, dst, bs.firstPriority) {
-			return
-		}
-		srcRate := srcLd.Loads[bs.firstPriority]
-		dstRate := dstLd.Loads[bs.firstPriority]
-		peerRate := peer.GetLoad(statistics.GetRegionStatKind(bs.rwTy, bs.firstPriority))
-		if srcRate-peerRate >= dstRate+peerRate {
+		// For write leader, only compare the first priority.
+		if bs.isBetterForWriteLeader() {
 			bs.cur.progressiveRank = -1
 		}
 	} else {
-		firstPriorityDimHot, firstPriorityDecRatio, secondPriorityDimHot, secondPriorityDecRatio := bs.getHotDecRatioByPriorities(srcLd, dstLd, peer)
-		greatDecRatio, minorDecRatio := bs.sche.conf.GetGreatDecRatio(), bs.sche.conf.GetMinorGreatDecRatio()
 		switch {
-		case firstPriorityDimHot && firstPriorityDecRatio <= greatDecRatio && secondPriorityDimHot && secondPriorityDecRatio <= greatDecRatio:
+		case bs.isBetter(bs.firstPriority) && bs.isBetter(bs.secondPriority):
 			// If belong to the case, two dim will be more balanced, the best choice.
-			if !bs.isTolerance(src, dst, bs.firstPriority) || !bs.isTolerance(src, dst, bs.secondPriority) {
-				return
-			}
 			bs.cur.progressiveRank = -3
```

Comment:
how about use enum to replace progressiveRank like -3, -2 , -1?

Reply:
we will add more case in another pr